Add Shipyard MCP beta

nadnad opened the pull request:
Model Context Protocol server for interacting with Shipyard. See readme for details on available commands etc.

nadnad committed:
Add Shipyard MCP beta

Model Context Protocol server for interacting with Shipyard.
See readme for details on available commands etc.

daniellle commented on pkg/mcp/errors/errors.go:
```diff
@@ -0,0 +1,239 @@
+package errors
```
maybe we could leave this file only for common usage and instantiation, and have new files for each error

`validation_error.go` (for ValidationError) and so on..

maxsokolovsky commented on commands/mcp.go:
```diff
@@ -0,0 +1,84 @@
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+
+	log.Println("MCP server running. Press Ctrl+C to stop.")
```
When I run `shipyard mcp serve` and then try to stop it with Ctrl+C (on macOS with zsh), I don't see any log messages about the process shutting down; it appears to hang.

nadnad replied:
Added commit to remedy this.

maxsokolovsky commented on pkg/mcp/schemas/environment.go:
```diff
@@ -0,0 +1,224 @@
+package schemas
```
Could these schemas be defined as type-safe structs rather than maps of strings to anything?

nadnad replied:
Wouldn't that be a quite big task to model out all of the JSON Schema vocabulary, including nested keywords etc? Or is there some JSON Schema library we can use for this?

maxsokolovsky commented on pkg/mcp/middleware/middleware.go:
```diff
@@ -0,0 +1,18 @@
+)
+
+// JSONRPCRequest represents a JSON-RPC 2.0 request for middleware processing
+type JSONRPCRequest struct {
```
This type is redefined in the `server` package.

nadnad replied:
Removed the duplicate definition

nadnad committed:
Fix MCP stdio transport deadlock on shutdown

Ensure Start allocates msg/err channels even for literal transports,
handle read errors via the buffered channels,
and cover the regression with a test.

nadnad committed:
server.py use middleware.JSONRPCRequest

Remove duplicate definition

nadnad committed:
Add instructions on adding MCP to Codex CLI

nadnad committed:
Note on where to find the Codex CLI config.toml file

nadnad committed:
Fix Claude Desktop: add prompts/list + notifications/initialized

Claude desktop calls them and errors out if not found

maxsokolovsky reviewed (approved):
One small thing: when Claude Desktop gets permission to use a tool for the first time, it works, but there is an error: `Unexpected token 'D', "DEBUG: MCP"... is not valid JSON`.
Since it's not blocking, perhaps it's fine for now.

Also, a few `shipyard` processes hang around on macOS, even after I fully shutdown the Claude Desktop app.

nadnad committed:
Merge branch 'main' into add-shipyard-mcp-beta

nadnad committed:
Remove debug statements
